rpcq/_server.py

```diff
@@ -74,17 +76,38 @@ def exit_handler(self, f: Callable):
         """
         self._connect(endpoint)
 
-        while True:
-            try:
-                # empty_frame may either be:
-                # 1. a single null frame if the client is a REQ socket
-                # 2. an empty list (ie. no frames) if the client is a DEALER socket
-                identity, *empty_frame, msg = await self._socket.recv_multipart()
-                request = from_msgpack(msg)
+        # spawn an initial listen task
+        listen_task = asyncio.ensure_future(self._socket.recv_multipart())
+        task_list = [listen_task]
 
-                asyncio.ensure_future(self._process_request(identity, empty_frame, request))
-            except Exception:
-                _log.exception('Exception thrown in Server run loop')
+        while True:
+            dones, pendings = await asyncio.wait(task_list, return_when=asyncio.FIRST_COMPLETED)
+            task_list = list(pendings)
+
+            for done in dones:
+                if done == listen_task:
+                    # empty_frame may either be:
+                    # 1. a single null frame if the client is a REQ socket
+                    # 2. an empty list (ie. no frames) if the client is a DEALER socket
+                    identity, *empty_frame, msg = done.result()
+                    request = from_msgpack(msg)
```

Comment:
`self._socket.recv_multipart` `from_msgpack` can raise.  The socket receive is unlikely but the `from_msgpack` happens whenever we change the message spec.  Currently it just spits out the issue to the server logs but and fails to inform the client but it doesn't take the server down

Reply:
I added some further error handling here so that (1) it catches socket processing / deser errors and (2) it resumes the server event loop in the case of an exception, so that the Server doesn't necessarily crash even with `serialize_exceptions` turned off.